### PR TITLE
Map protocol codes

### PR DESCRIPTION
**Pull request by ramon-bernardo**
- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
- Map all packet codes between client-server.

Thanks otclient.

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -137,6 +137,7 @@ set(tfs_HDR
 	${CMAKE_CURRENT_LIST_DIR}/position.h
 	${CMAKE_CURRENT_LIST_DIR}/protocolgame.h
 	${CMAKE_CURRENT_LIST_DIR}/protocol.h
+    ${CMAKE_CURRENT_LIST_DIR}/protocolcodes.h
```
Comment (EPuncker) [inline src/CMakeLists.txt:140]:
```suggestion
	${CMAKE_CURRENT_LIST_DIR}/protocolcodes.h
```

File: src/protocolgame.cpp [hunk 2]
```diff
@@ -2449,16 +2449,16 @@ void ProtocolGame::sendCreatureTurn(const Creature* creature, uint32_t stackpos)
...
-		msg.add<uint16_t>(0xFFFF);
+		msg.add<uint16_t>(0xFFFF); // use creature id instead position
 		msg.add<uint32_t>(creature->getID());
 	} else {
 		msg.addPosition(creature->getPosition());
 		msg.addByte(stackpos);
 	}
 
-	msg.add<uint16_t>(0x63);
+	msg.addByte(GAME_SERVER_SEND_CLIENT_CHECK);
```
Comment (ramon-bernardo) [inline src/protocolgame.cpp:2461]: Otclient does not handle the rest of this packet.
it seems to me that this rest is `GAME_SERVER_SEND_CLIENT_CHECK`.

We don't have a turn packet at otclient 😞

File: src/protocolcodes.h [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+	// all in game opcodes must be equal or greater than 50
+	CLIENT_FIRST_GAME_OPCODE = 50,
+
+	// otclient ONLY
+	CLIENT_EXTENDED_OPCODE = 50,
+	CLIENT_CHANGE_MAP_AWARE_RANGE = 51,
+
+	// NOTE: add any custom opcodes in this range (51 - 99)
+
+	// original tibia ONLY
```
Comment (EPuncker) [inline src/protocolcodes.h:221]:
```suggestion
	// official client ONLY
```

File: src/protocolcodes.h [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+
+	// NOTE: add any custom opcodes in this range (51 - 99)
+	GAME_SERVER_CHANGE_MAP_AWARE_RANGE = 51,
+	GAME_SERVER_ATTCHED_EFFECT = 52,
+	GAME_SERVER_DETACH_EFFECT = 53,
+	GAME_SERVER_CREATURE_SHADER = 54,
+	GAME_SERVER_MAP_SHADER = 55,
+	GAME_SERVER_FLOOR_DESCRIPTION = 75,
+
+	// original tibia ONLY
```
Comment (EPuncker) [inline src/protocolcodes.h:64]:
```suggestion
	// official client ONLY
```

File: src/protocolcodes.h [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+	GAME_SERVER_SEND_RESTING_AREA_STATE = 169,
+	GAME_SERVER_TALK = 170,
+	GAME_SERVER_CHANNELS = 171,
+	GAME_SERVER_OPEN_CHANNEL = 172,
+	GAME_SERVER_OPEN_PRIVATE_CHANNEL = 173,
+	GAME_SERVER_RULE_VIOLATION_CHANNEL = 174,
+	GAME_SERVER_RULE_VIOLATION_REMOVE = 175,
+	GAME_SERVER_EXPERIENCE_TRACKER = 175,
+	GAME_SERVER_RULE_VIOLATION_CANCEL = 176,
+	GAME_SERVER_RULE_VIOLATION_LOCK = 177,
```
Comment (ramon-bernardo) [inline src/protocolcodes.h:134]: Violations need to be mapped?

`GAME_SERVER_EXPERIENCE_TRACKER` = `GAME_SERVER_RULE_VIOLATION_REMOVE`

**Review by Zbizu — CHANGES_REQUESTED** [review]

**Comment by luanluciano93** [comment]
@ramon-bernardo Can you share your discord?

**Comment by luanluciano93** [comment]
@ramon-bernardo It's a good time to reopen this, my friend.

**Comment by ramon-bernardo** [comment]
@luanluciano93 Im rewriting the connection and protocol, so we won't need this anymore 😄 (https://github.com/ramon-bernardo/fork-forgottenserver-otland/pull/1).

But first, we want to merge the smart pointer things

**Pull request closed without merging**